backend/src/controllers/authController.ts

```diff
@@ -40,35 +39,41 @@ export const register = async (req: any, res: Response) => {
     } else {
       newUser = { ...result, name: result.name.trim().replace(/\s+/g, " ") };
     }
+
     const createdUser: any = await User.create(newUser);
-    sendEmail(
-      createdUser.name,
-      createdUser.email,
-      "Welcome " + createdUser.name,
-      generateMessage(
+    if(createdUser) {
+      await sendEmail(
         createdUser.name,
         createdUser.email,
-        createdUser.role,
-        password,
-      ),
-    ).catch((error) => console.error(error));
+        "Welcome " + createdUser.name,
+        generateMessage(
+          createdUser.name,
+          createdUser.email,
+          createdUser.role,
+          password,
+        ),
+      ).catch((error) => console.error(error));
```

Comment:
Could you return this error in response instead of consoling it

Reply:
I think in this block, the response object is not reacheable. Since I used console.error not console.log it makes sense. console. error() writes to stderr , whereas console. log() writes to stdout.